### PR TITLE
mpu9250 improvements

### PR DESCRIPTION
**Pull request by dakejahl**
Changed mag orientation to match gyro and accel. Corrected temperature calculation. Removed 'check_registers' function from the cycle loop. Deleted stale comments.

### CODE REVIEW

File: src/drivers/imu/mpu9250/mag.cpp [hunk 1]
```diff
@@ -209,14 +209,14 @@ MPU9250_mag::_measure(struct ak8963_regs data)
...
 	 */
-	mrb.x_raw =  data.x;
-	mrb.y_raw = -data.y;
+	mrb.x_raw =  data.y;
+	mrb.y_raw =  data.x;
 	mrb.z_raw = -data.z;
 
-	float xraw_f =  data.x;
-	float yraw_f = -data.y;
+	float xraw_f =  data.y;
```
Comment (dagar) [inline src/drivers/imu/mpu9250/mag.cpp]: Any idea what was going on here? I wasn't sure if it was an attempt to conform with other external mags like 3dr unit.
@mhkabir @dakejahl

Comment (AlexKlimaj) [reply src/drivers/imu/mpu9250/mag.cpp]: The idea behind this is to bring the mag into the same reference frame as the accel and gyro. 

image

Comment (dagar) [reply src/drivers/imu/mpu9250/mag.cpp]: I meant the existing mapping that clearly doesn't match the datasheet. This is the common orientation for external mags. https://github.com/PX4/Firmware/blob/master/src/drivers/magnetometer/hmc5883/hmc5883.cpp#L980-L983

Comment (LorenzMeier) [reply src/drivers/imu/mpu9250/mag.cpp]: @dagar It was designed to match the orientation on Pixhawk. Not necessarily the best approach in terms of maintainability.

Comment (mcsauder) [reply src/drivers/imu/mpu9250/mag.cpp]: To fix the driver for the generic case it's going to force each user to reset CAL_MAG0_ROT and recalibrate.  It seems important that this driver should be corrected to match the datasheet, both for the temperature and for the orientation.  Is it possible to slate this for the 2.0.0 release and (unfortunately) force users who upgrade to accept the backwards breakage for the mpu9250 along the lines of what I have suggested to #9054?

Comment (dagar) [reply src/drivers/imu/mpu9250/mag.cpp]: Something still doesn't make sense here. How is this functional at all for existing pixracer or pixhawk 2.1 users? In both cases they have an internal mpu9250 no additional mag rotation applied.

Comment (AlexKlimaj) [reply src/drivers/imu/mpu9250/mag.cpp]: @dagar It looks like the calibration routine is applying negative scalars to the values and making it work.

Comment (mcsauder) [reply src/drivers/imu/mpu9250/mag.cpp]: It works when the calibration applies negative scalars.

Comment (mcsauder) [reply src/drivers/imu/mpu9250/mag.cpp]: Our mpu9250 sensor board orientation matches pixracer precisely.

Comment (dagar) [reply src/drivers/imu/mpu9250/mag.cpp]: I don't see how negative scale factors would account for swapping x and y without an additional rotation error, but then why aren't accel and gyro a problem? Something else would have to be wrong for this to make sense.


Here's a stock pixracer with default rotation.

`sensors status` level facing south
screenshot from 2018-05-21 20-32-33

`sensors status` nose down
screenshot from 2018-05-21 20-33-21

mag calibration (no negative scale factors)
screenshot from 2018-05-21 20-34-04

No issues with gyro or accel inconsistencies either.

Comment (mcsauder) [reply src/drivers/imu/mpu9250/mag.cpp]: @dagar, see my edited comment above. You are correct in that it indeed requires a change to the driver startup rotation to correct the system in its entirety.

**Pull request closed without merging**